### DISCUSSION TITLE
Capture outer scope variable in config function

### DISCUSSION THREAD

**Discussion by Tastyep** [wbthomason/packer.nvim#513] [Q&A] [answered]
Hello,

I'm trying to capture a variable into a lambda function, but the variable is always nil.
Am I forced to use a global variable to achieve that, or is there another solution ? Thanks in advance.

Minimal example:
``` lua
local my_var = 1

use({
"some-plugin",
config = function()
  print("VAR: ", my_var) -- <-- prints nil
end
})
```

**Comment by wbthomason** [comment, marked as answer]
Unfortunately, this is a well-known limitation of `string.dump` in Lua, which cannot capture closure variables. If you need this kind of capture, you can use a global, or you can pass a string instead of a function to the `config` key (e.g. `[[require(config.some-plugin)]]`, where `~/.config/nvim/lua/config/some-plugin.lua` is a file that does whatever setup you want. Forthcoming changes in #402 will remove this limitation, as we will no longer need to use `string.dump`.

**Reply by Tastyep** [reply]
Alright, thank you for the details :+1:
